docs/aws/audit/esmonitoring/rules/es_ri_recent_purchases.mdx

```diff
@@ -23,6 +23,252 @@ CBP
+
+4. Evaluate the recent purchases based on your organization's usage patterns and needs. Check if the instance type, size, and term match your requirements. If there are any discrepancies or unusual purchases, it might indicate a misconfiguration or an oversight.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by following the instructions provided in the AWS CLI User Guide.
+
+2. List all Elasticsearch domains: Use the following AWS CLI command to list all the Elasticsearch domains in your AWS account.
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all Elasticsearch domains in your AWS account.
+
+3. Describe Elasticsearch domain: For each domain listed in the previous step, use the following AWS CLI command to get detailed information about the domain.
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name <DomainName>
+   ```
+   Replace `<DomainName>` with the name of the Elasticsearch domain. This command will return detailed information about the specified Elasticsearch domain, including its configuration.
+
+4. Review Reserved Instance purchases: In the output of the previous command, look for the `ReservedElasticsearchInstanceOfferingId` field. This field indicates the ID of the Reserved Instance offering that was used to purchase the domain. If this field is present and not empty, it means that a Reserved Instance was recently purchased for the domain. You should review this purchase to ensure that it was intentional and is cost-effective.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+from datetime import datetime, timedelta
+```
+
+2. Create a session using your AWS credentials. Replace 'your_access_key', 'your_secret_access_key', and 'your_region' with your actual AWS credentials and region.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='your_access_key',
+    aws_secret_access_key='your_secret_access_key',
+    region_name='your_region'
+)
+```
+
+3. Create an Elasticsearch client:
+
+```python
+es_client = session.client('es')
+```
+
+4. Now, you can use the `describe_reserved_elasticsearch_instances` method to get information about your reserved instances. You can filter the results to only include instances purchased in the last 30 days.
+
+```python
+response = es_client.describe_reserved_elasticsearch_instances()
+
+recent_purchases = []
+for instance in response['ReservedElasticsearchInstances']:
+    purchase_date = instance['StartTime']
+    if purchase_date >= datetime.now() - timedelta(days=30):
+        recent_purchases.append(instance)
+
+print(recent_purchases)
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>